File: host1/webapp1/static/webapp1/tic-tac-toe/v2/playground_equipment.js
/**
 * 遊具
 */
class PlaygroundEquipment {
    constructor() {
        // あとで onStart(...) を呼出してください
    }

    /**
     * 対局開始時
     *
     * @param {string} myPiece - "X", "O", "_"
     */
    onStart(myPiece) {
        console.log(`[PlaygroundEquipment onStart] myPiece=${myPiece} PC_EMPTY=${PC_EMPTY} PC_X_LABEL=${PC_X_LABEL}`);

        // 盤面
        this._board = new Board();

        // 棋譜
        this._record = new Record();

        // 自分の手番
        this._myTurn = new MyTurn(myPiece);

        // 「相手の手番に着手しないでください」というアラートの可視性
        this._isVisibleAlertWaitForOther = false;

        // ゲームオーバー状態
        this._gameoverState = new GameoverSet(GameoverSet.none);
    }

    /**
     * 盤
     */
    get board() {
        return this._board;
    }

    /**
     * 棋譜
     */
    get record() {
        return this._record;
    }

    /**
     * 自分のターン
     */
    get myTurn() {
        return this._myTurn;
    }

    /**
     * ゲームオーバー状態
     */
    get gameoverState() {
        return this._gameoverState;
    }

    /**
     * マスがすべて埋まっていますか
     */
    isBoardFill() {
        return this.record.length == 9;
    }

    /**
     * 同じ駒が３個ありますか
     */
    isThere3SamePieces() {
        return 5 <= this.record.length;
    }

    /**
     * 「相手の手番に着手しないでください」というアラートの可視性
     */
    get isVisibleAlertWaitForOther() {
        return this._isVisibleAlertWaitForOther;
    }

    set isVisibleAlertWaitForOther(value) {
        this._isVisibleAlertWaitForOther = value;
    }
}

File: host1/webapp1/views/tic_tac_toe/v2/playground_equipment.py
from webapp1.views.tic_tac_toe.v2 import game_rule
#    ------- --------------------        ---------
#    1       2                           3
# 1. アプリケーション フォルダー名
# 2. ディレクトリー名
# 3. Python ファイル名。拡張子抜き


class PlaygroundEquipment():
    """遊具"""

    def __init__(self):
        # あとで onStart(...) を呼出してください
        pass

    def onStart(self, myPiece):
        """対局開始時

        Parameters
        ----------
        myPiece : str
            "X", "O", "_"
        """

        self._board = [game_rule.PC_EMPTY, game_rule.PC_EMPTY, game_rule.PC_EMPTY, game_rule.PC_EMPTY,
                       game_rule.PC_EMPTY, game_rule.PC_EMPTY, game_rule.PC_EMPTY, game_rule.PC_EMPTY, game_rule.PC_EMPTY]
        """盤面"""

        self._countOfMove = 0
        """何手目"""

        self._isMyTurn = myPiece == game_rule.PC_X_LABEL
        """自分の手番か（初回は先手）"""

        self._isVisibleAlertWaitForOther = False
        """「相手の手番に着手しないでください」というアラートの可視性"""

        self._gameoverState = game_rule.GAMEOVER_NONE
        """ゲームオーバーしてません"""

        # イベントハンドラはそのまま

    def getPieceBySq(self, sq):
        """盤上のマス番号で示して、駒を取得

        Parameters
        ----------
        sq : int
            マス番号

        Returns
        -------
        _type_
            _description_
        """
        return self._board[sq]

    def setPiece(self, sq, piece):
        """盤上のマスに駒を上書きします

        Parameters
        ----------
        sq : int
            マス番号
        piece : str
            駒
        """
        self._board[sq] = piece

    def incrementCountOfMove(self):
        """手数を１増やします"""
        self._countOfMove += 1

    def isBoardFill(self):
        """マスがすべて埋まっていますか"""
        return self._countOfMove == 9

    def isThere3SamePieces(self):
        """同じ駒が３個ありますか"""
        return 5 <= self._countOfMove

    @property
    def isMyTurn(self):
        """私のターンですか"""
        return self._isMyTurn

    @isMyTurn.setter
    def isMyTurn(self, value):
        self._isMyTurn = value

    @property
    def isVisibleAlertWaitForOther(self):
        """「相手の手番に着手しないでください」というアラートの可視性"""
        return self._isVisibleAlertWaitForOther

    @isVisibleAlertWaitForOther.setter
    def isVisibleAlertWaitForOther(self, value):
        self._isVisibleAlertWaitForOther = value

    @property
    def gameoverState(self):
        """ゲームオーバー状態"""
        return self._gameoverState

    @gameoverState.setter
    def gameoverState(self, value):
        self._gameoverState = value
